userland/libtock/crt0.c

```diff
@@ -47,8 +47,9 @@ void _start(void* text_start,
             void* app_heap_break __attribute__((unused))) {
 
   // Allocate stack and data. `brk` to STACK_SIZE + got_size + data_size +
-  // bss_size from start of memory
-  uint32_t stacktop = (uint32_t)mem_start + STACK_SIZE;
+  // bss_size from start of memory. The stack must be aligned to a multiple of
+  // 4 so enforce that here.
+  uint32_t stacktop = (uint32_t)mem_start + (STACK_SIZE & 0xFFFFFFFC);
```

Comment:
Does this method possibly shrink the requested STACK_SIZE by up to 4 (or soon 8) bytes? Feels like we should err on being too large rather than too small.

Reply:
Also, I don't think we have any guarantees on the alignment of mem_start here? Should this thus be something like `(mem_start + STACK_SIZE + 0x7) % 0xFFFF_FFF8` ?